src/groupTypes.ts

```diff
@@ -70,32 +68,40 @@ export interface Deserializer<T> {
 
 export interface SerializationHelpers {
     desElt(bytes: Uint8Array): Elt
+
     desScalar(bytes: Uint8Array): Scalar
 
     eltDes: Deserializer<Elt>
     scalarDes: Deserializer<Scalar>
 
     eltSize(compressed?: boolean): number
+
     scalarSize(): number
 }
 
 export interface Group extends SerializationHelpers {
     id: GroupID
-    size: number
 
     newScalar(): Scalar
+
     newElt(): Elt
+
     identity(): Elt
+
     generator(): Elt
+
     mulGen(s: Scalar): Elt
+
     randomScalar(): Promise<Scalar>
+
     hashToGroup(msg: Uint8Array, dst: Uint8Array): Promise<Elt>
+
     hashToScalar(msg: Uint8Array, dst: Uint8Array): Promise<Scalar>
+
+    hash(hashID: HashID, input: Uint8Array): Promise<Uint8Array>
```

Comment:
Yep, I realized that, but figured it was a pragmatic choice given the only current way we can configure things is via setting the Oprf.Group object. Notably, the hash method does take a hashID parameter.


What do you suggest as an alternative?

Reply:
The issue here is that Web Crypto doesn't support SHAKE256 and thus we need to provide an implementation, but we are polluting the Group interface in the process. I didn't really like it at first but learned to live with it given the `hash(hashId, input)` signature. 

Were you ever planning to extract a group-ts repository and update the interfaces here to perhaps support consuming a Suite instance (Group, hash) as well as enum ? 

```typescript
export function getKeySizes(id: SuiteID): { Nsk: number; Npk: number } {
    const gg = Oprf.getGroup(id)
    return { Nsk: gg.scalarSize(), Npk: gg.eltSize(true) }
}

export function validatePrivateKey(id: SuiteID, privateKey: Uint8Array): boolean {
    try {
        const s = Oprf.getGroup(id).desScalar(privateKey)
        return !s.isZero()
    } catch (_) {
        return false
    }
}
```

i.e. `suite: SuiteID | Suite`

I think at this point if there's a desire to separate the hash provider it might be worth stepping back a bit and see which way you really want to take it all?

Reply:
How about making the hash() method a static method of the GroupCons? 
I guess it's slightly less icky